### PR TITLE
#164489909 user reading stats

### PR DESCRIPTION
**Pull request by blaisebakundukize**
**What does this PR do?**

This PR set and get user reading stats

**Description of Task to be completed?**

setting users reading stats happens when they read an article, and for getting them they have to view their profile.

**How should this be manually tested?**

- Login to application
- read an article to set your reading stats
- visit your profile to view your total reading stats

**What are the relevant pivotal tracker stories?**

[#164489909](https://www.pivotaltracker.com/story/show/164489909)

**Screenshots (if appropriate)**

Screen Shot 2019-06-10 at 12 19 59

Screen Shot 2019-06-10 at 12 22 52

Screen Shot 2019-06-11 at 12 06 31

### CODE REVIEW

File: src/components/articles/SingleArticle.js [hunk 1]
```diff
@@ -77,22 +85,22 @@ export class SingleArticle extends Component {
...
-    prevPath, userId, articleId2
-  } = this.state;
-  const { articles: { article, error, message } } = this.props;
-  if (article !== null) if (article && article.article !== undefined) single = article.article;
-  return (
+  render() {
+    let single;
+    const {
+      prevPath, userId, articleId2
+    } = this.state;
```
Comment (zellymiriam) [inline src/components/articles/SingleArticle.js]: Add empty line

File: src/components/articles/SingleArticle.js [hunk 2]
```diff
@@ -77,22 +85,22 @@ export class SingleArticle extends Component {
...
+    const { articles: { article, error, message } } = this.props;
+    if (article !== null) if (article && article.article !== undefined) single = article.article;
```
Comment (nakiwuge) [inline src/components/articles/SingleArticle.js:102]: Add new line before this line

File: src/components/users/profile/ViewProfile.js [hunk 3]
```diff
@@ -48,120 +51,122 @@ export class ViewProfile extends Component {
   render() {
     let successMessage;
     const {
-      loading, profile, followers, following, articles, message
+      loading, profile, followers, following, articles, message, readingStats, errorGetReadingStats
     } = this.props;
     const { startLoading } = this.state;
     if (message !== '') setTimeout(() => { window.location.reload(true); }, 500);
+    let displayReadingStats = null;
+    if (errorGetReadingStats === null && readingStats > 0) {
```
Comment (nakiwuge) [inline src/components/users/profile/ViewProfile.js:61]: Add new line before this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@blaisebakundukize good job on the implementation however for greater user experience add a hover effect with `Read articles` on the icon you used for reading.

Also, check out the comments I left on the PR.

**Comment by blaisebakundukize** [comment]
@zellymiriam, thank you for the feedback. I am looking forward to taking action on them. Thanks

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**